affectedmoduledetector/src/main/kotlin/com/dropbox/affectedmoduledetector/AffectedModuleDetector.kt

```diff
@@ -113,8 +111,39 @@ abstract class AffectedModuleDetector {
         private const val ENABLE_ARG = "affected_module_detector.enable"
         var isConfigured = false
 
+        private var affectedModuleConfiguration: AffectedModuleConfiguration? = null
+
+        @JvmStatic
+        fun getRootConfiguration(project: Project): AffectedModuleConfiguration {
+            if (affectedModuleConfiguration == null) {
+                var rootProject = project
+                while (!rootProject.isRoot || rootProject.parent != null) {
+                    rootProject = rootProject.parent!!
+                }
+
+                affectedModuleConfiguration = requireNotNull(
+                    value = rootProject.extensions.findByType(AffectedModuleConfiguration::class.java),
+                    lazyMessage = {
+                        "Root project $rootProject must have the ${AffectedModuleConfiguration.name} extension added."
+                    }
+                )
+            }
+
+            return affectedModuleConfiguration!!
+        }
+
+        @JvmStatic
+        fun getTestConfiguration(project: Project): AffectedTestConfiguration {
```

Comment:
getting of root configuration and test configuration should be in one place with one contract. It should be the same. 

Now u can get any configuration from AMD file and it is clear for understanding and DRY also

Reply:
`AffectedTestConfiguration` only is used internally in the plugin so should exist inside the plugin class. There is no real benefit to exposing this as I'm not sure why another plugin would ever want to call this. Making it public and static means this could get called by another plugin without having initialized the plugin which just fail since this is non-nullable. Can you provide a usecase for getting this value outside the plugin? (tests don't count as we could just as easily have this helper function internal in the tests if we needed it)

Also, there is very little usecase for getting the configuration as well which is why its only called once as well before this change. You can get what you need from the instance of the `AffectedModuleDetector` in most cases